File: examples/interactive_segmentation/android/shared/src/main/java/com/google/mediapipe/tasks/core/Quantization.java
package com.google.mediapipe.tasks.core;

public enum Quantization {
  FLOAT_16("fp16"),
  FLOAT_32("fp32"),
  INT_8("int8");

  private final String description;

  /**
   * Factory method to create a {@link Quantization} from the model metadata.
   *
   * @param description a string representing the quantization type.
   */
  public static Quantization fromCanonicalName(String description) {
    for (Quantization quantization : Quantization.values()) {
      if (quantization.getDescription().equals(description)) {
        return quantization;
      }
    }
    throw new IllegalArgumentException("Unsupported quantization type: " + description);
  }

  Quantization(String description) {
    this.description = description;
  }

  public String getDescription() {
    return description;
  }
}

File: vision-provider/src/main/kotlin/com/google/mediapipe/tasks/core/Quantization.kt
package com.google.mediapipe.tasks.core

enum class Quantization(description: String) {
    FLOAT16("fp16"),
    FLOAT32("fp32"),
    INT8("int8");

    val description: String?

    init {
        this.description = description
    }

    companion object {
        /**
         * Factory method to create a [Quantization] from the model metadata.
         *
         * @param description a string representing the quantization type.
         */
        fun fromCanonicalName(description: String?): Quantization {
            for (quantization in values()) {
                if (quantization.description == description) {
                    return quantization
                }
            }
            throw IllegalArgumentException("Unsupported quantization type: " + description)
        }
    }
}
